madmom/io/audio.py

```diff
@@ -315,16 +312,56 @@ def decode_to_pipe(infile, fmt='f32le', sample_rate=None, num_channels=1,
                         replaygain_mode=replaygain_mode,
                         replaygain_preamp=replaygain_preamp)
     # redirect stdout to a pipe and buffer as requested
-    if isinstance(infile, (Signal, io.IOBase)):
+    if isinstance(infile, (Signal, file_types)):
         proc = subprocess.Popen(call, stdin=subprocess.PIPE,
                                 stdout=subprocess.PIPE, bufsize=buf_size)
     else:
         proc = subprocess.Popen(call, stdout=subprocess.PIPE, bufsize=buf_size)
     return proc.stdout, proc
 
 
+def nonstreamable_mp4_file_object(infile, cmd="ffprobe"):
+    """
+    Test if the given audio file object is a non-streamable MPEG4 container.
+    Decode the given audio and return a file-like object for reading the
+    samples, as well as a process object.
+
+    Parameters
+    ----------
+    infile : file_types
+        File object to decode.
+    cmd : {'ffprobe', 'avprobe'}, optional
+        Probing command (defaults to ffprobe, alternatively supports avprobe).
+
+    Returns
+    -------
+    boolean
+        True when audio is non-streamable container, False otherwise.
+
+    """
+    call = [cmd, "-v", "debug", "-hide_banner",
+            "-show_entries", "format=format_name",
+            "-print_format", "default=nokey=1:noprint_wrappers=1",
+            "pipe:0"]
+    proc = subprocess.Popen(call, stdin=subprocess.PIPE,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+    out, err = proc.communicate(infile.read())
+    retcode = proc.poll()
+    infile.seek(0)
+    if retcode:
+        raise subprocess.CalledProcessError(retcode, call, output=err)
+    container_format = out.decode().strip()
+    partial_file = "partial file" in err.decode()
+    if container_format == "mov,mp4,m4a,3gp,3g2,mj2" and partial_file:
+        return True
+    else:
+        return False
+
+
 def decode_to_memory(infile, fmt='f32le', sample_rate=None, num_channels=1,
-                     channel=None, skip=None, max_len=None, cmd='ffmpeg',
+                     channel=None, skip=None, max_len=None,
+                     cmd_decode='ffmpeg', cmd_probe='ffmpeg',
```

Comment:
Is `ffmpeg` the correct default for `cmd_probe`? The docstring suggests that `ffprobe` should be used.